### PR TITLE
Improve the auth checking code for passsword-resets

### PR DESCRIPTION
**Pull request by orta**
Also checks for 401s from auth to inform the user about failed credentials.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  These files have changed: Artsy/Networking/API_Modules/ARUserManager.m
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems">Danger Swift</a> against 1fe2089781e8c6fba7a4c6f02292fa49a605015b
</p>

**Comment by orta** [comment]
err, thanks Danger swift?

**Comment by l2succes** [comment]
👍

File: Artsy/Networking/API_Modules/ARUserManager.m [hunk 1]
```diff
@@ -210,16 +210,16 @@ - (void)loginWithUsername:(NSString *)username
...
+                    authenticationFailure(error);
+                }
+            }];
+            [userOp start];
         }
 
-        failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error, id JSON) {
-        if (JSON) {
+    failure:^(NSURLRequest *request, NSHTTPURLResponse *response, NSError *error, id JSON) {
+        if ([response statusCode] == 401 || JSON) {
```
Comment (l2succes) [inline Artsy/Networking/API_Modules/ARUserManager.m]: 👍

File: Artsy/Networking/API_Modules/ARUserManager.m [hunk 2]
```diff
@@ -210,16 +210,16 @@ - (void)loginWithUsername:(NSString *)username
...
             if (authenticationFailure) {
```
Comment (alloy) [inline Artsy/Networking/API_Modules/ARUserManager.m]: Should this really even be optional?

**Pull request merged**